grails-app/conf/UrlMappings.groovy

```diff
@@ -25,6 +26,12 @@ class UrlMappings {
             controller = "stockMovement"
         }
 
+        "/stockRequest/$action/$id**?" {
+            controller = "stockMovement"
+            isStockRequest = true
+            expectedSourceType = RequisitionSourceType.ELECTRONIC
```

Comment:
I don't think adding `expectedSourceType == RequisitionSourceType.PAPER` in the `stockMovement` URL would help, because again, we might come to situation when user tries to delete `stockRequest` using `stockMovement`'s URL and just passes expectedSourceType "by hand", so I don't know what would be the best solution there. 
With that said I realized that my approach also wasn't the best, because without checking actual URL I also allowed user to delete `stockRequest` using `stockMovement` URL - the user would just need to pass `isStockRequest` and `expectedSourceType` and it would work (I think nothing wrong would actually happen, but the user should get the message, that he is trying to do something bad).

Reply:
>  we might come to situation when user tries to delete stockRequest using stockMovement's URL and just passes expectedSourceType "by hand"

Correct, but there's a possibility that hardcoding the parameter in the UrlMapping would override the value passed in the request parameters. We should test that out.

> With that said I realized that my approach also wasn't the best,

Correct. We need to inspect the URL. Which is why I want to push this logic into the filters (it doesn't really belong up at the controller level).